server/pubEdge/model.js

```diff
@@ -0,0 +1,35 @@
+export default (sequelize, dataTypes) => {
+	return sequelize.define(
+		'PubEdge',
+		{
+			pubId: { type: dataTypes.UUID, allowNull: false },
+			targetForeignPublication: { type: dataTypes.UUID, allowNull: true },
+			targetPubId: { type: dataTypes.UUID, allowNull: true },
+			relationType: { type: dataTypes.STRING, allowNull: false },
+			rank: { type: dataTypes.TEXT, allowNull: false },
```

Comment:
It was obviously intentional based on your unit tests, but just out of curiosity, why is `rank` represented w/ a string rather than a numeric type like `INTEGER` ?

Reply:
The short answer (which will become longer as I keep typing) is because we need to be able to reorder things with `rank`, and moving the item with `rank = 100` to the beginning of the list means incrementing the `rank` of 99 other items (the first item is now the second, etc). No doubt there is some Postgres-fu to make this happen with a single query (and I would like to learn it!) but the limiting case still means updating hundreds or thousands of rows.

So historically we've used a scheme where `rank` (FKA `order`) is a `DOUBLE`, and when item C is inserted between items B and A, we set `C.order = (A.order + B.order) / 2`. But we run out of precision pretty fast. So these days, we use strings to create ordered collections in Postgres because they have infinite precision — given two strings `a < c` and the lexical ordering relation `<`, you can always slot a `b` in between them such that `a < b < c` just by choosing `b = a + (some characters)`. We have a library that uses a more intelligent algorithm to pick `b` for us, and it tends to work great!